sdk/servicebus/service-bus/src/serviceBusAtomManagementClient.ts

```diff
@@ -891,4 +894,19 @@ export class ServiceBusAtomManagementClient extends ServiceClient {
     const ruleResponse: RuleResponse = Object.assign(rule || {}, { _response: response });
     return ruleResponse;
   }
+
+  private buildResponse(fn: Function, response: HttpOperationResponse): any {
+    try {
+      return fn(response);
+    } catch (err) {
+      log.error(`Failure parsing response from service - ${err}`);
+      throw new RestError(
+        `Response from service is in bad format`,
```

Comment:
core-http does actually include the inner error message, so while I think it's a bit misleading/possibly confusing, let's align completely with core-http's error message:

`Error "${err}" occurred while parsing the response body`

Reply:
Updated

Reply:
A few places below, @ramya-rao-a proposed a different wording. We should unify the error message format for anything throwing the PARSE_ERROR error code. I propose the following modifications:

```js
`Error "${error}" occurred while parsing the response body - ${details}`
```

* `error` is the inner error where it might be helpful to the user, otherwise omitted (e.g. in cases where its only JS TypeError/etc. that might be thrown).
* details provides additional context, including the word "service" if the service is likely at fault.


So, elsewhere, we have the following error string:

```js
`Response from service is in bad format. Expected response to be in Atom XML format.`
```

Based on above guidelines, this could be updated to:

```js
`Error occurred while parsing the response body - expected the service to return a response body in Atom XML format`
```

Thoughts @ramya-rao-a @ramya0820?

Reply:
I like the `Error occurred while parsing the response body - expected the service to return a response body .... ` as the common first part of the error message.


> error is the inner error where it might be helpful to the user, otherwise omitted 

So in our case, what kind of errors would make sense to surface up? From what I could see, all of the cases fall under "log the actual issue, but throw a generic issue" kind of a case.

Reply:
I looked over all the errors coming from parsing service output, and I see only 3 types of cases. In all cases, I believe we should throw a generic error with no inner error. The inner error should only be logged as it is not actionable by the user.
- Failure to parse xml out of string. 
    - This error comes from the `parseXml()` method in core-http
    - Recommended error message: `Error occurred while parsing the response body - expected the service to return valid xml content.`
- Failure to find feed or entry from the parsed xml
    - This error comes from the `parseAtomResult()`
    - Recommended error message: `Error occurred while parsing the response body - expected the service to return xml content with either feed or entry elements.` 
- Failure to deserialize expected shape of sql parameters, authorization rules, count details etc
    - This error comes from the various helper methods we have that get called in the `deserialize()` step
    - Recommended error message: `Error occurred while parsing the response body - cannot form a queue/rule/subscription/topic object using the response from the service.`
    - The one catch here is that we have to use the right word among queue/rule/subscription/topic and so cannot use a single generic message for all cases.

@ramya0820 Please sync up with @bterlson to get a confirmation on the above before making changes.

Reply:
Am fine omitting the inner error, I was mostly slavishly aligning with core-http but I still agree it's not useful. Your suggestions seem good to me!

Reply:
We discussed about adding in "atom" to the second error message.
Good with other suggestions, updated PR.

Reply:
Looks good.